utils/biospecimen.js

```diff
@@ -472,6 +472,39 @@ const biospecimenAPIs = async (req, res) => {
             return res.status(500).json({message: 'Error removing bag', code:500});
         }
     }
+    else if (api === 'getSpecimensByCollectionIds'){
+        if(req.method !== 'GET') {
+            return res.status(405).json(getResponseJSON('Only GET requests are accepted!', 405));
+        }
+
+        const collectionIdArray = req.query.collectionIdArray.split(',');
+        const isBPTL = req.query.isBPTL ?? false;
+        if (!collectionIdArray) return res.status(400).json(getResponseJSON('CollectionIdArray is missing.', 400));
```

Comment:
Check `collectionIdArray` before processing it.

Reply:
Updated.